app/controllers/deliveries_controller.rb

```diff
@@ -0,0 +1,76 @@
+class DeliveriesController < ApplicationController
+  def index
+    matching_deliveries = Delivery.all
+
+    @list_of_deliveries = matching_deliveries.order({ :created_at => :desc })
+
+    render({ :template => "deliveries/index" })
+  end
+
+  def show
+    the_id = params.fetch("path_id")
+
+    matching_deliveries = Delivery.where({ :id => the_id })
+
+    @the_delivery = matching_deliveries.at(0)
+
+    render({ :template => "deliveries/show" })
+  end
+
+  def create
+    the_delivery = Delivery.new
+    the_delivery.description = params.fetch("query_description")
+    the_delivery.supposed_to_arrive_on = params.fetch("query_supposed_to_arrive_on")
+    the_delivery.details = params.fetch("query_details")
+    the_delivery.user_id = current_user.id
+    the_delivery.arrived = params.fetch("query_arrived", false)
```

Comment:
This works, but you also could set it to false because usually when a delivery is created, it has not arrived yet so it can automatically be set to false